tensorflow/python/keras/layers/preprocessing/preprocessing_stage.py

```diff
@@ -94,3 +96,120 @@ def map_fn(x):
                                              reset_state=reset_state)
 
 
+class FunctionalPreprocessingStage(base_preprocessing_layer.PreprocessingLayer,
+                                   functional.Functional):
+  """A functional preprocessing stage.
+
+  This preprocessing stage wraps a graph of preprocessing layers into a
+  Functional-like object that enables you to `adapt()` the whole graph via
+  a single `adapt()` call on the preprocessing stage.
+
+  Arguments:
+    inputs: List of input tensors (must be created via `tf.keras.Input()`).
```

Comment:
Can inputs and outputs be dicts instead of lists? (from the test it seems like the answer is yes)

Reply:
Thanks for pointing out, I will update the docstring.